src/types/middleware.ts

```diff
@@ -0,0 +1,28 @@
+import { EffectScope } from 'vue'
+import { MaybePromise } from '@/types/utilities'
+
+type Route = {
+  name: string,
+  params: Record<string, unknown>,
+  query: unknown,
+  hash: string,
+}
+
+interface RouteState {
```

Comment:
This is intentionally empty. The idea is that users can declare their own RouteState at implementation via declaration merging. This will likely need a namespace or module to work properly.

Reply:
does this need to be exported to work the way we want it to?

Reply:
Yeah probably exported as part of a ts module or namespace. Don't 100% have my head wrapped around that. Just waited to nail down the internal types.

Reply:
I think we just need to export it and then users can add their custom types like this
```typescript
declare module '@kitbag/router {
  export interface RouteState {
    user: User,
    account: Account,
  }

  export interface RouteRejection {
    Auth: unknown
  }
}
```

So I've updated these to be exported.